google/cloud/security/common/util/log_util.py

```diff
@@ -12,29 +12,90 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-"""A basic util that wraps logging."""
+"""A basic util that wraps logging.
+
+Setup logging for Forseti Security based on deployment environment and
+flags passed to the binaries.
+
+If flags are passed, determine logging behavior as follows:
+  - For --use-cloud-logging, use Stackdriver/Cloud Logging (if available),
+    otherwise fall back to local logging.
+  - For --nouse-cloud-logging, only use local logging.
+
+If there are no flags passed to the Forseti tools, default logging behavior
+is as follows:
+  - Stackdriver/Cloud Logging, if querying metadata server succeeds. This means
+    that Forseti is running on Compute Engine.
+  - Otherwise, fall back to local logging.
+"""
 
 import logging
-import os
-
-class LogUtil(object):
-    """Utility to wrap logging setup."""
-
-    @classmethod
-    def setup_logging(cls, module_name):
-        """Setup logging configuration.
-
-        Args:
-            module_name: The name of the module to describe the log entry.
-        """
-        formatter = logging.Formatter(
-            '%(asctime)s %(name)-12s %(levelname)-8s %(message)s')
-        handler = logging.StreamHandler()
-        handler.setFormatter(formatter)
-        my_logger = logging.getLogger(module_name)
-        my_logger.addHandler(handler)
-        if os.getenv('DEBUG'):
-            my_logger.setLevel(logging.DEBUG)
-        else:
-            my_logger.setLevel(logging.INFO)
-        return my_logger
+
+import gflags as flags
+
+from google.cloud.security.common.gcp_api import cloud_logging
+from google.cloud.security.common.gcp_api import compute
+
+FLAGS = flags.FLAGS
+flags.DEFINE_boolean('use_cloud_logging', False,
+                     'Use Cloud Logging, if available.')
+flags.DEFINE_boolean('nouse_cloud_logging', False, 'Do not use Cloud Logging.')
+flags.DEFINE_boolean('debug', False, 'Show DEBUG level logs.')
+
+LOG_FORMAT = '%(asctime)s %(name)-12s %(levelname)-8s %(message)s'
+MODULE_PREFIX = 'google.cloud.security'
+
+def get_logger(module_name):
+    """Setup the basic logger.
+
+    Args:
+        module_name: The name of the mdule to describe the log entry.
+
+    Returns:
+        An instance of the configured logger.
+    """
+    formatter = logging.Formatter(LOG_FORMAT)
+    console_handler = logging.StreamHandler()
```

Comment:
Should we default to SysLogHandler instead? Or maybe use logging.config.fileConfig and include a default logging configuration that can be overridden by changing the configuration file rather than changing the code?